beacon-chain/db/state.go

```diff
@@ -178,12 +217,21 @@ func (db *BeaconDB) SaveCurrentAndFinalizedState(beaconState *pb.BeaconState) er
 	}
 
 	db.currentState = currentState
+
+	// Delete historical states if we are saving a new finalized state.
+	if err := db.deleteHistoricalStates(); err != nil {
+		return err
+	}
 	return db.update(func(tx *bolt.Tx) error {
 		chainInfo := tx.Bucket(chainInfoBucket)
 		beaconStateEnc, err := proto.Marshal(beaconState)
 		if err != nil {
 			return err
 		}
+		// Putting in finalized state.
+		if err := chainInfo.Put(stateLookupKey, beaconStateEnc); err != nil {
```

Comment:
Shouldn't we only be saving a finalized state if it is finalized? This seems to be saving the current state as finalized. @terenc3t can you confirm?

Reply:
Agreed. I think through out this PR, there's a few places where `historical state` and `finalized state` are mixed up

Reply:
it's probably fine to eagerly save this state and consider some pruning mechanism later.

Reply:
@rauljordan This method saves both the current and finalized state together , we need it in initial sync when we receive the finalized state from the peer. If not we have to do two separate db writes together.